resolver/resolver.go

```diff
@@ -167,11 +170,37 @@ type BuildOptions struct {
 	Dialer func(context.Context, string) (net.Conn, error)
 }
 
+// An Endpoint is one network endpoint, or server, which may have multiple
+// addresses with which it can be accessed.
+type Endpoint struct {
+	// Addresses contains a list of addresses used to access this endpoint.
+	Addresses []Address
+
+	// Attributes contains arbitrary data about this endpoint intended for
+	// consumption by the LB policy.
+	Attributes *attributes.Attributes
```

Comment:
Should we name it `BalancerAttributes` instead, just for future proofing.

Reply:
I would prefer just "Attributes" since no subchannel can or would ever see this field.

If someone wanted to add an attributes at this level for the subchannel to see (I guess pick-first would merge it into the address attributes for each address?) I would rather have the producer set it for all the addresses individually instead.